src/aggregation/bucket/filter.rs

```diff
@@ -174,60 +184,84 @@ impl PartialEq for FilterAggregation {
 /// Document evaluator for filter queries
 /// This avoids running separate query executions and instead evaluates queries per document
 struct DocumentQueryEvaluator {
-    /// Cached weight for the current segment
-    weight: Option<Box<dyn Weight>>,
-    /// Note: We can't pre-create the scorer because it maintains state (current doc position)
-    /// that would be invalid across multiple document evaluations.
-    /// Instead, we cache the SegmentReader which is cheap (Arc-based internally).
-    segment_reader: Option<SegmentReader>,
+    /// The scorer for document matching
+    /// We create this once per segment and reuse it for all document checks.
+    /// This is critical for performance.
+    scorer: Option<Box<dyn Scorer>>,
```

Comment:
This is initialized in `new`, and then never removed, afaict? Does it need to be an `Option`?